Objects/obmalloc.c

```diff
@@ -493,7 +499,9 @@ set_up_allocators_unlocked(PyMemAllocatorName allocator)
         set_allocator_unlocked(PYMEM_DOMAIN_MEM, &malloc_alloc);
         set_allocator_unlocked(PYMEM_DOMAIN_OBJ, &malloc_alloc);
 
-        if (allocator == PYMEM_ALLOCATOR_MALLOC_DEBUG) {
+        int is_debug = (allocator == PYMEM_ALLOCATOR_MALLOC_DEBUG);
+        _PyRuntime.allocators.is_debug_enabled = is_debug;
```

Comment:
Are these assignments necessary?  It seems like this is going to get set in `set_up_debug_hooks_unlocked` anyway.  I suppose there's a question if you were to change the allocator after initializing it, but it seems like that would blow up horribly anyway?

Reply:
I think they're not necessary - I'll remove them and make sure things still work.

Reply:
Turns out they are needed. We may statically initialize `is_debug_enabled` to `1` in a debug build, but then disable debug hooks via configuration (such as, `PYTHONMALLOC=mimalloc`). In that case we need to clear  `is_debug_enabled`.